### PR TITLE
feat: improve Prometheus endpoint stability and add test coverage

### PR DESCRIPTION
**Pull request by aqibmohammadkhan**
- Ensure graceful handling when Prometheus registry is unavailable.
- Prevent double semaphore releases in ConcurrencyLimitHandler.
- Add unit and integration tests for Prometheus endpoint and concurrency handler.

**Issue**

https://github.com/gravitee-io/issues/issues/APIM-12316

**Description**

This PR fixes a semaphore leak on the Prometheus scrape endpoint that could progressively block /_node/metrics/prometheus with 429 Too Many Requests after repeated scrape failures/timeouts.

 - Fixed concurrency slot release in ConcurrencyLimitHandler by:
      - handling closeHandler in addition to bodyEndHandler / exceptionHandler
      - guarding release with AtomicBoolean to ensure permits are released exactly once even if multiple handlers fire.
 - Hardened PrometheusEndpoint response lifecycle by:
      - using explicit SafeBufferedWriter + BufferedWriter resources
      - closing/ending responses with ended() + closed() guards to avoid invalid response state transitions

 - Improved Prometheus registry resolution in PrometheusEndpoint:
      - resolves and stores PrometheusMeterRegistry once in constructor
      - supports direct PrometheusMeterRegistry, CompositeMeterRegistry, and graceful fallback (501) when unavailable

 - Why this change
      - On scrape timeout (Timeout while waiting for write queue to drain), the endpoint could close the response without triggering the previously wired release path, causing semaphore permits to leak. After enough occurrences (e.g., limit of 3), the endpoint permanently rejected scrapes until restart.

 - Test coverage
      - Added/expanded tests to validate both correctness and regression scenarios:
           - ConcurrencyLimitHandlerTest
                - permit release on success, exception, and close paths
                - release-only-once when multiple handlers fire
                - concurrency-limit behavior under repeated requests
                - regression scenario for repeated close events without permanent rejection
            
           - PrometheusEndpointTest
                - happy path scrape
                - IO exception handling
                - already-ended/already-closed response behavior
                - header/chunked response behavior
                - composite registry and missing-registry handling
            
            - PrometheusEndpointIntegrationTest
                 - end-to-end scrape output
                 - repeated scrape stability
                 - repeated scrape stability with concurrency limit enabled
                 - disabled endpoint behavior

 - Expected outcome
       - Gateway nodes should no longer accumulate stuck concurrency slots after scrape failures/timeouts, and Prometheus scraping should continue to recover normally without requiring a node restart.

**Additional context**

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `7.6.0-APIM-12316-Prometheus-scrapping-error-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/node/gravitee-node/7.6.0-APIM-12316-Prometheus-scrapping-error-SNAPSHOT/gravitee-node-7.6.0-APIM-12316-Prometheus-scrapping-error-SNAPSHOT.zip)

### CODE REVIEW

File: gravitee-node-management/src/test/java/io/gravitee/node/management/http/metrics/prometheus/PrometheusEndpointIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+            .compose(req -> req.send())
+            .onComplete(ar -> {
+                if (ar.succeeded()) {
+                    ar.result().body().onComplete(bodyAr -> statusFuture.complete(ar.result().statusCode()));
+                } else {
+                    statusFuture.completeExceptionally(ar.cause());
+                }
+            });
+        return statusFuture.get(10, TimeUnit.SECONDS);
+    }
```
[the remark this answers is not in the document]
Comment (aqibmohammadkhan) [reply gravitee-node-management/src/test/java/io/gravitee/node/management/http/metrics/prometheus/PrometheusEndpointIntegrationTest.java:178]: the current implementation is kept as-is because it explicitly waits for body consumption before asserting status and is already stable in this suite.

File: gravitee-node-management/src/test/java/io/gravitee/node/management/http/metrics/prometheus/PrometheusEndpointIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+                    response.handler(chunk -> bodyBuilder.append(chunk.toString()));
+                    response.endHandler(v -> bodyFuture.complete(bodyBuilder.toString()));
+                    response.exceptionHandler(bodyFuture::completeExceptionally);
+                } else {
+                    bodyFuture.completeExceptionally(ar.cause());
+                }
+            });
+
+        return await(Future.fromCompletionStage(bodyFuture));
+    }
```
[the remark this answers is not in the document]
Comment (aqibmohammadkhan) [reply gravitee-node-management/src/test/java/io/gravitee/node/management/http/metrics/prometheus/PrometheusEndpointIntegrationTest.java:203]: we intentionally keep manual handler/endHandler aggregation. Using response.body() caused intermittent empty-body behavior in the concurrency-limited Prometheus scrape path, while explicit streaming ensures we complete only on endHandler and preserves test determinism.

Given these tests validate concurrency/lifecycle edge cases, we prefer explicit response-stream handling over a more concise async style here.

**Review by remibaptistegio — APPROVED** [review]
